UI/src/model/locationSearch.js

```diff
@@ -96,25 +96,39 @@
       });
     };
 
+
+    var extractSearchResultInfo = function (massTransitStop) {
+      return { lon: _.get(massTransitStop, 'lon'),
+        lat: _.get(massTransitStop, 'lat'),
+        id: _.get(massTransitStop, 'id')
+      };
+    };
+
     /**
      * Mass transit stop national id search, combined with road number search
      *
      * @param input
      * @returns {*}
      */
     var roadNumberAndNationalIdSearch = function(input) {
-      return $.when(backend.getMassTransitStopByNationalIdForSearch(input.text), backend.getCoordinatesFromRoadAddress(input.text)).then(function(result,roadData) {
-        var returnObject = roadLocationAPIResultParser(roadData);
-         if (_.get(result[0], 'success')) {
-          var lon = _.get(result[0], 'lon');
-          var lat = _.get(result[0], 'lat');
+      return $.when(backend.getMassTransitStopByNationalIdForSearch(input.text) ).then(function(result) {
```

Comment:
Why we delete the code  "backend.getCoordinatesFromRoadAddress(input.text)" ?

Reply:
Because the input is only the national ID and VKM require tie, osa, etaisyys value to return values.
Otherwise it will return HTTP 500

Reply:
Ok seems fine, I will resolve this thread.